app/models/repo.rb

```diff
@@ -3,4 +3,14 @@ class Repo < ApplicationRecord
 
   validates :full_name, presence: true
   validates :name, presence: true
+
+  scope :ordered_by_active, -> { order("active desc") }
+
+  def toggle_active
+    if active?
```

Comment:
You can replace this entire method with just: `update_attributes(active: !active?)`